### PR TITLE
Feat/scout 24 comp age categories

### PR DESCRIPTION
**Pull request by dejwi**
[SCOUT-24](https://playmakerpro.atlassian.net/browse/SCOUT-24?atlOrigin=eyJpIjoiZDBiM2IxNTI4YmE4NGI0MjhmNjdjYzZkZDllNzQzNGYiLCJwIjoiaiJ9)

### Task Description
Competition age categories
- table
- create/edit form
- translations
- index/create/edit/id pages
- admin role check
- add to admin panel

### Additional Notes (optional)
- no filter form as there's literally no filtering for competition age categories
- commits duplicated after my merge attempt:(

### CODE REVIEW

File: public/locales/pl/common.json [hunk 1]
```diff
@@ -80,6 +80,8 @@
   "COMPETITIONS": "Rozgrywki",
   "COMPETITION_GROUP": "Grupa rozgrywkowa",
   "COMPETITION_GROUPS": "Grupy rozgrywkowe",
+  "COMPETITION_AGE_CATEGORY": "Kategoria wiekowa rozgrywki",
```
Comment (jakubfiglak) [inline public/locales/pl/common.json:83]: It should be `Kategoria wiekowa rozgrywek` I suppose:)

File: src/components/icons/index.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 export {
   PermContactCalendar as AdminIcon,
+  Category,
```
Comment (jakubfiglak) [inline src/components/icons/index.ts:3]: I think we should reexport this as `CategoryIcon` for consistency

File: src/modules/competition-age-categories/forms/utils.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import * as yup from 'yup'
+
+import { CreateTeamDto } from '@/modules/teams/types'
+
+import {
+  CompetitionAgeCategortyDto,
+  UpdateCompetitionAgeCategoryDto,
+} from '../types'
+
+export const initialValues: CreateTeamDto = {
```
Comment (jakubfiglak) [inline src/modules/competition-age-categories/forms/utils.ts:12]: I think it's a leftover from another file, we should adjust it to match `CreateCompetitionAgeCategoryDto` :)

File: src/modules/competition-age-categories/forms/utils.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  groupId: 0,
+  minut90url: '',
+  transfermarktUrl: '',
+  lnpId: '',
+}
+
+export function generateCreateValidationSchema(t: TFunction) {
+  return yup
+    .object({
+      name: yup.string().required(t('teams:NO_NAME_ERROR')),
```
Comment (jakubfiglak) [inline src/modules/competition-age-categories/forms/utils.ts:25]:
```suggestion
      name: yup.string().required(t('comp-age-categ:NO_NAME_ERROR')),
```

There's wrong i18n namespace used here and translations don't work properly

Screenshot 2022-08-09 at 20 40 30

Please adjust:)

**Review by jakubfiglak — CHANGES_REQUESTED** [review]
There are some minor issues to be addressed (please see my comments), but other than that, as usual, fantastic work 🚀 

We're gonna need to adjust to some backend changes - as you can see there's no `list` endpoint for this module & regular `GET /` endpoint acts as a list (there's no pagination or sorting available). This is something I need to fix and make consistent throughout entire API and when it's done, we will adjust frontend code (I will create separate ticket for that)

File: src/modules/competition-age-categories/forms/utils.ts (hunk 4 above)
Comment (dejwi) [reply src/modules/competition-age-categories/forms/utils.ts]: added separatly with other changes and accepting suggestion and merge is blocked:c

**Review by jakubfiglak — APPROVED** [review]

**Pull request merged**